datadog_lambda/trigger.py

```diff
@@ -0,0 +1,211 @@
+        log_group = logs.get("logGroup", "cloudwatch")
+        return "arn:{}:logs:{}:{}:log-group:{}".format(
+            aws_arn, region, account_id, log_group
+        )
+
+    # e.g. arn:aws:events:us-east-1:123456789012:rule/my-schedule
+    if source == "cloudwatch-events" and event.get("resources"):
+        return event.get("resources")[0]
+
+
+def get_event_source_arn(source, event, context):
+    event_source_arn = event.get("eventSourceARN") or event.get("eventSourceArn")
+
+    event_record = get_first_record(event)
+    if event_record:
+        event_source_arn = event_record.get("eventSourceARN") or event_record.get(
+            "eventSourceArn"
+        )
+
+    if event_source_arn is None:
+        event_source_arn = parse_event_source_arn(source, event, context)
+
+    return event_source_arn
+
+
+def extract_http_tags(event):
+    """
+    Extracts HTTP facet tags from the triggering event
+    """
+    http_tags = {}
+    request_context = event.get("requestContext")
+    path = event.get("path")
+    method = event.get("httpMethod")
+    if request_context and request_context.get("stage"):
+        if request_context.get("domainName"):
+            http_tags["http.url"] = request_context["domainName"]
+
+        path = request_context.get("path")
+        method = request_context.get("httpMethod")
+        # Version 2.0 HTTP API Gateway
+        apigateway_v2_http = request_context.get("http")
+        if event.get("version") == "2.0" and apigateway_v2_http:
+            path = apigateway_v2_http.get("path")
+            method = apigateway_v2_http.get("method")
+
+    if path:
+        http_tags["http.url_details.path"] = path
+    if method:
+        http_tags["http.method"] = method
+
+    headers = event.get("headers")
+    if headers and headers.get("Referer"):
+        http_tags["http.referer"] = headers["Referer"]
+
+    return http_tags
+
+
+def extract_trigger_tags(event, context):
+    """
+    Parses the trigger event object to get tags to be added to the span metadata
+    """
+    trigger_tags = {}
+    event_source = parse_event_source(event)
+    if event_source:
+        trigger_tags["trigger.event_source"] = event_source
+
+        event_source_arn = get_event_source_arn(event_source, event, context)
+        if event_source_arn:
+            trigger_tags["trigger.event_source_arn"] = event_source_arn
+
+    if event_source in ["api-gateway", "application-load-balancer"]:
+        trigger_tags.update(extract_http_tags(event))
+
+    return trigger_tags
+
+
+def set_http_status_code_tag(span, response):
+    """
+    If the Lambda was triggered by API Gateway or ALB add the returned status code
+    as a tag to the function execution span.
+    """
+    is_http_trigger = span and span.get_tag("trigger.event_source") == (
+        "api-gateway" or "application-load-balancer"
+    )
+    if not is_http_trigger:
+        return
+
+    status_code = "200"
+    if response is None:
```

Comment:
What actually happens when a Lambda function returns `None` as response to API Gateway? Does that actually cause API Gateway to return 502?

Reply:
I think you assumed that `response` is `None` only if the function throws an exception [here](https://github.com/DataDog/datadog-lambda-python/pull/101/files#diff-5efc779a7e3c73ebd812eb62ff5dce68db1e0ed9ef9765bb076638733e8c5ae6R110)? But what if the customer's handler function somehow does return `None` as response?

Reply:
Yup, good thought -- it seems that if the Lambda handler returns `None`, API Gateway will always return a 502 code. I think that relates to the "incompatible output" here: https://docs.aws.amazon.com/apigateway/api-reference/handling-errors/